cgc/kmeans.py

```diff
@@ -112,62 +114,84 @@ def compute(self):
         # Search for value k
         silhouette_avg_list = np.array(
             [])  # List of average silhouette measure of each k value
-        kmeans_cc_list = []
+        kmean_cluster_list = []
         for k in self.k_range:
             # Compute Kmean
-            kmeans_cc = KMeans(n_clusters=k, max_iter=self.kmean_max_iter).fit(
-                self.stat_measures_norm)
+            kmean_cluster = KMeans(n_clusters=k,
+                                   max_iter=self.kmean_max_iter).fit(
+                                       self.stat_measures_norm)
             silhouette_avg = silhouette_score(self.stat_measures_norm,
-                                              kmeans_cc.labels_)
+                                              kmean_cluster.labels_)
             silhouette_avg_list = np.append(silhouette_avg_list,
                                             silhouette_avg)
-            kmeans_cc_list.append(kmeans_cc)
+            kmean_cluster_list.append(kmean_cluster)
         idx_k = np.argmax(silhouette_avg_list)
+        if np.sum(silhouette_avg_list == silhouette_avg_list[idx_k]) > 1:
+            idx_k_list = np.argwhere(
+                silhouette_avg_list == silhouette_avg_list[idx_k]).reshape(
+                    -1).tolist()
+            logger.warning(
+                "Multiple potential k found: {}, picking the smallest one: {}".
```

Comment:
Maybe a bit more explicit:
```suggestion
                "Multiple k values with the same silhouette score: {},"
                "picking the smallest one: {}".
```